### ISSUE TITLE
Inline block styles not applied in pop-out preview panel

### ISSUE THREAD

**Issue by Andrew-Starr** [WordPress/gutenberg#39944]
### Description

When registering custom block styles with `inline_style` styling, the style is not applied to the iframe inside of the block styles preview panel.

### Step-by-step reproduction instructions

Add a custom block style with inline styling to any block theme:

```
register_block_style(
	'core/paragraph',
	array(
		'name' => 'just-a-test',
		'label' => 'Just a Test',
		'inline_style' => '.is-style-just-a-test{color: #ffcc00;font-weight: 900;}'
	)
);
```

In the post editor select a paragraph and in Styles hover over the 'Just a Test' style button.
The pop-out block style preview will appear, but notice that the custom style is not applied to the preview paragraph within the panel.
The required class `is-style-just-a-test` is correctly added to the paragraph element in the preview, but not the corresponding inline style.

Now add the custom style to the theme stylesheet and notice that the style is applied correctly to the preview panel.

So it seems there is an issue only with inline styles not being applied within the iframe of the preview panel.

Now deactivate Gutenberg and the styles previews are displayed all at once in the sidebar without the pop-out preview panel. In this scenario the inline styles are applied correctly.

### Environment info

WP 5.9.2
Gutenberg 12.9.0

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by glendaviesnz** [comment]
I was able to replicate this:

https://user-images.githubusercontent.com/3629020/163094969-8d50a24d-d12d-49c2-bbbe-ce9cb5a7b9fb.mp4

**Comment by t-hamano** [comment]
I was able to reproduce it as reported in #43250.
I will describe the results of that test here as well.

---

## Step-by-step reproduction instructions

- Defines the following code in `functions.php`:
```php
register_block_style(
	'core/button',
	array(
		'name'         => 'test',
		'label'        => 'Test',
		'inline_style' => '.wp-block-button.is-style-test { border: 2px red solid }',
	)
);
```

- Open the post editor and insert the button block.
- Apply "Test" style and confirm that a red border has been applied to the button.
- Confirm that the style disappears when you change to mobile tablet view.
- Confirm that the same problem is occurred in the site editor.

## Screenshot

### ✅ Frontend

![frontend](https://user-images.githubusercontent.com/54422211/184826603-535c5995-06d5-4977-847e-05e9521070ca.png)

### ✅ Post Editor

![post-editor](https://user-images.githubusercontent.com/54422211/184826669-625546ba-2f14-497b-8457-c3272f7c5ad3.png)

### ❌Post Editor (Mobile/Tablet View)

![post-editor-mobile](https://user-images.githubusercontent.com/54422211/184826858-735c5ed6-90ea-4117-b447-1d3191ea4a24.png)

### ❌ Site Editor

![site-editor](https://user-images.githubusercontent.com/54422211/184826903-6172b64a-bc9c-4878-bd64-28cf3735baa5.png)

## Inline CSS Details

### ✅ Post Editor

The style defined in `inline_css` is correctly added as `wp-block-library-inline-css`.

![post-editor-css](https://user-images.githubusercontent.com/54422211/184827272-05f7b41a-0d16-4c5b-86c4-c638c00b4842.png)

### ❌ Site Editor

Styles defined in `inline_css` aren't added to `wp-block-library-inline-css`.

![site-editor-css](https://user-images.githubusercontent.com/54422211/184827427-f1d8f944-31e8-4fc3-afea-537a1c19956a.png)

### Environment info

- WordPress Version: 6.0.1
- Gutenberg Version: 13.9.0-rc.3
- OS: Windows 11

**Comment by dashkevych** [comment]
I was also able to reproduce this issue in WordPress **6.1-beta2-54361** with active Gutenberg plugin (**14.1.1**).

**Comment by ellatrix** [comment]
This has been fixed in WP 6.3.


<img width="679" alt="Screenshot 2023-08-17 at 13 43 34" src="https://github.com/WordPress/gutenberg/assets/4710635/542faddb-3631-4340-b513-5dfab08537e6">
